src/components/menu/Menu.tsx

```diff
@@ -64,17 +57,15 @@ const Menu = <S,>({
   selected,
   ...others
 }: MenuProps<S>): ReactElement<any, any> | null => {
-  const [stateWidth, setStateWidth] = useState<number | undefined>(0);
-  const [stateHeight, setStateHeight] = useState<number | undefined>(0);
-  const [statePosition, setPosition] = useState<string | undefined>(position);
+  const [positionState, setPositionState] = useState<Position>(position);
```

Comment:
PP: changing `position` after the component was mounted won't do anything. Either name it something like `initialPosition` to clarify or add a `useEffect` to update the state when the prop changes.